libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -425,22 +427,31 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         }
         hasInstance = true
 
-        navigator = NavigationComponentProvider.createNativeNavigator(
+        val config = NavigatorLoader.createConfig(
             navigationOptions.deviceProfile,
             navigatorConfig,
+        )
+        historyRecorders = NavigatorLoader.createHistoryRecorders(
```

Comment:
Can we extract this invocation and
```
historyRecorder.historyRecorderHandle = historyRecorders.historyRecorderHandle
copilotHistoryRecorder.historyRecorderHandle = historyRecorders.copilotHistoryRecorderHandle
```
into a private method? Now it's copy-pasted in `recreateNavigatorInstance`.